ete2/parser/newick.py

```diff
@@ -237,6 +242,21 @@ class as the root(This allows to work with custom TreeNode
 
 def _read_newick_from_string(nw, root_node, format):
     """ Reads a newick string in the New Hampshire format. """
+    # begin: Quoted text is mapped to references
+    quoteMap={}
+    unquoted_nw=''
```

Comment:
Use `unquoted_nw = []` and append to it. Then `''.join(unquoted_nw)` once you are done.
String concatenation `+=` in a loop is too inefficient.